stellargraph/utils/version_validation.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Data61, CSIRO
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import warnings
+from packaging import version
+from ..version import __version__
+
+__all__ = ["validate_notebook_version"]
+
+
+def validate_notebook_version(notebook_version):
```

Comment:
Is it worth having a test for this? Something like

```python
def test_current_version_older():
    with pytest.raises(ValueError, match=".* requires StellarGraph version 987.654.321, but an older version .*/1172"):
        validate_notebook_version("987.654.321")

def test_current_version_newer():
    with pytest.warns(DeprecationWarning, match=".* is designed for an older StellarGraph version 0.0.0 and may not function correctly with the newer .*/1172"):
        validate_notebook_version("0.0.0")
```

An optional extension would be:

```python
def test_current_version_vs_release():
    # compare a current pre-release version A.B.Cb0 (or similar) against A.B.C
    match = re.match(r"^(\d+\.\d+\.\d+)[^\d].*$")
    if not match:
        pytest.skip(f"can only test release versions for prereleases, not {stellargraph.__version__}")

    release_version = match[0]
    with pytest.raises(ValueError, match=".* requires StellarGraph version .* but an older version"):
        validate_notebook_version(relase_version)
```

If we were feeling very enthusiastic, we'd make `__version__` pluggable to be able to test `1.2.3b0` vs. `1.2.3` easily, but I think it's probably ok with just the above

Reply:
Yes, worth having tests - I forgot!  I've made a simpler dev/release test using patch.